amplipi/eeprom.py

```diff
@@ -76,166 +131,123 @@ class UnsupportedFormatError(Exception):
-      raise UnsupportedFormatError("Unsupported format")
-    self._write_number(4, SERIAL_ADDR, serial)
-
-  def write_unit_type(self, unit_type: UnitType) -> None:
-    """Write unit type to EEPROM."""
-    if self.get_format() not in SUPPORTED_FORMATS:
-      raise UnsupportedFormatError("Unsupported format")
-    self._write(UNIT_TYPE_ADDR, [unit_type.value])
-
-  def write_board_type(self, board_type: BoardType) -> None:
-    """Write board type to EEPROM."""
-    if self.get_format() not in SUPPORTED_FORMATS:
-      raise UnsupportedFormatError("Unsupported format")
-    self._write(BOARD_TYPE_ADDR, [board_type.value])
-
-  def write_board_rev(self, rev: Tuple[int, str]) -> None:
-    """Write board rev to EEPROM."""
-    if self.get_format() not in SUPPORTED_FORMATS:
-      raise UnsupportedFormatError("Unsupported format")
-    self._write_number(1, BOARD_REV_ADDR, rev[0])
-    self._write_letter(BOARD_REV_ADDR + 1, rev[1].upper())
-
-  def get_format(self) -> int:
-    """Check EEPROM format."""
-    fmt = self._read(FORMAT_ADDR, 1)[0]
-
-    return fmt
-
-  def get_serial(self) -> int:
-    """Get serial from EEPROM."""
-    if self.get_format() not in SUPPORTED_FORMATS:
-      raise UnsupportedFormatError("Unsupported format")
-
-    return self._read_number(4, SERIAL_ADDR)
-
-  def get_unit_type(self) -> UnitType:
-    """Get unit type from EEPROM."""
-    if self.get_format() not in SUPPORTED_FORMATS:
-      raise UnsupportedFormatError("Unsupported format")
-
-    return UnitType(self._read(UNIT_TYPE_ADDR, 1)[0])
-
-  def get_board_type(self) -> BoardType:
-    """Get board type from EEPROM."""
-    if self.get_format() not in SUPPORTED_FORMATS:
-      raise UnsupportedFormatError("Unsupported format")
-
-    return BoardType(self._read(BOARD_TYPE_ADDR, 1)[0])
+  def read_board_info(self, unit_addr=0x08) -> Optional[BoardInfo]:
+    """Get board info from EEPROM."""
 
-  def get_board_rev(self) -> Tuple[int, str]:
-    """Get board rev from EEPROM."""
-    if self.get_format() not in SUPPORTED_FORMATS:
+    # Read page from EEPROM
+    # TODO: Handle exceptions?
+    try:
+      eeprom_data = self._read(0, 9, unit_addr)
+      # Convert to BoardInfo if successful
+      board_info = BoardInfo(eeprom_data)
+    except:
+      return None   # If there is no EEPROM, such as on an old AmpliPi, read should return nothing
+    if board_info.format not in SUPPORTED_FORMATS:
       raise UnsupportedFormatError("Unsupported format")
+    return board_info
 
-    return (self._read_number(1, BOARD_REV_ADDR),
-            self._read_letter(BOARD_REV_ADDR + 1))
-
-  def get_board_info(self) -> BoardInfo:
-    """Get board info from EEPROM."""
-    return BoardInfo(self.get_serial(),
-                     self.get_unit_type(),
-                     self.get_board_type(),
-                     self.get_board_rev())
-
+  # TODO: Do we need this? If so implement like the _read() function above. If not remove.
```

Comment:
If we remove this, it probably makes sense to keep this pretty comment somewhere